jira/resilientsession.py

```diff
@@ -304,45 +308,27 @@ def __recoverable(
                     "Response headers for ConnectionError are only printed for log level DEBUG."
                 )
 
-        if isinstance(response, Response):
-            if response.status_code in [429]:
-                is_recoverable = True
-                number_of_tokens_issued_per_interval = response.headers.get(
-                    "X-RateLimit-FillRate"
-                )
-                token_issuing_rate_interval_seconds = response.headers.get(
-                    "X-RateLimit-Interval-Seconds"
-                )
-                maximum_number_of_tokens = response.headers.get("X-RateLimit-Limit")
-                retry_after = response.headers.get("retry-after")
-                msg = f"{response.status_code} {response.reason}"
-                warning_msg = "Request rate limited by Jira."
-
-                warning_msg += (
-                    f" Request should be retried after {retry_after} seconds.\n"
-                    if retry_after is not None
-                    else "\n"
-                )
-                if (
-                    number_of_tokens_issued_per_interval is not None
-                    and token_issuing_rate_interval_seconds is not None
-                ):
-                    warning_msg += f"{number_of_tokens_issued_per_interval} tokens are issued every {token_issuing_rate_interval_seconds} seconds.\n"
-                if maximum_number_of_tokens is not None:
-                    warning_msg += (
-                        f"You can accumulate up to {maximum_number_of_tokens} tokens.\n"
-                    )
-                warning_msg = (
-                    warning_msg
-                    + "Consider adding an exemption for the user as explained in: "
-                    + "https://confluence.atlassian.com/adminjiraserver/improving-instance-stability-with-rate-limiting-983794911.html"
-                )
+        elif isinstance(response, Response):
+            recoverable_error_codes = [
+                HTTPStatus.TOO_MANY_REQUESTS,
+                HTTPStatus.SERVICE_UNAVAILABLE,
+            ]
 
-                LOG.warning(warning_msg)
+            if response.status_code in recoverable_error_codes:
+                retry_after = response.headers.get("Retry-After")
+                if retry_after:
```

Comment:
this looks like regression: 429 should be still always retried

Reply:
429s are retried, just with a different delay as per previous behaviour -- see line 322